### PR TITLE
Wizard recipe: Minuit2-v6.15.1

### PR DESCRIPTION
**Pull request by jstrube**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: Minuit2
* Version: v6.15.1

@staticfloat please review and merge.

### CODE REVIEW

File: M/Minuit2/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Minuit2"
+version = v"6.15.1"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/jstrube/Minuit2.git", "32b4337fab6aca8cde5a0e18dfd9e05130eb7184"),
```
Comment (giordano) [inline M/Minuit2/build_tarballs.jl]: What's the upstream source?

Comment (jstrube) [reply M/Minuit2/build_tarballs.jl]: Minuit2 is part of the ROOT distribution, but there's no separate package. There's a CMake file in the minuit dir of the ROOT source tree that let's you build a self-contained library without pulling in the rest of ROOT. I just put that dir into its own repo to make easier to get the code if you don't want the rest of ROOT.

Comment (giordano) [reply M/Minuit2/build_tarballs.jl]: So if I wanted to submit a patch upstream where I should look?

Comment (jstrube) [reply M/Minuit2/build_tarballs.jl]: I had to look, and amazingly, it's actually on GitHub now... https://github.com/root-project/root. One could probably change the build script to download the whole tree and still only build minuit2. That would be easier for keeping things in sync.

**Comment by giordano** [comment]
```
[23:36:45] In file included from /workspace/srcdir/root-6-18-04/math/minuit2/../mathcore/inc/Math/IOptions.h:15:0,
[23:36:45]                  from /workspace/srcdir/root-6-18-04/math/minuit2/../mathcore/inc/Math/GenAlgoOptions.h:15,
[23:36:45]                  from /workspace/srcdir/root-6-18-04/math/mathcore/src/GenAlgoOptions.cxx:13:
[23:36:45] /workspace/srcdir/root-6-18-04/math/minuit2/../mathcore/inc/Math/Error.h:73:20: fatal error: TError.h: No such file or directory
[23:36:45] compilation terminated.
```
is the same error I was getting when trying to build for macOS with Clang.  It looks like we need to work this out in any case.

**Comment by jstrube** [comment]
Yeah, I'll have to try and find my notes. I think I manually added some files when making the repo...

**Comment by giordano** [comment]
> I think I manually added some files when making the repo...

This contradicts the [building instructions](https://github.com/root-project/root/tree/195edc0e84eee988f529aa72665a78f8cc2cf82a/math/minuit2#building):

> To build, use the standard CMake procedure; on most systems, this looks like:
> ```
> mkdir PATH_TO_MINIUT2_BUILD
> cd PATH_TO_MINUIT2_BUILD
> cmake PATH_TO_MINUIT2_SOURCE
> cmake --build .
> ```

But I wouldn't be surprised it they weren't accurate...

**Comment by jstrube** [comment]
OK, so in my repo I had hacked a sourcefile to just `define` a macro to use the right includes for standalone. Here, I am passing this macro to CMake. Not sure if this method translates to Clang, though.
Additionally, I need to hack the CMake file (should probably be a proper patch, but it's only one line) to create a shared lib instead of a static one.

Not sure what this latest issue is, though. I didn't think I touched the `cd` command.

**Comment by jstrube** [comment]
oh. Sorry. I'm no help at all.

**Comment by giordano** [comment]
It looks like it worked

**Comment by giordano** [comment]
For the record, the patch I wrote for macOS has been merged upstream: https://github.com/root-project/root/pull/5016.

**Comment by jstrube** [comment]
Awesome. Thank you

**Pull request merged**